x/dex/keeper/core.go

```diff
@@ -385,7 +383,7 @@ func (k Keeper) CancelLimitOrderCore(goCtx context.Context, msg *types.MsgCancel
 		return types.ErrNotEnoughLimitOrderShares
 	}
 
-	tranche := NewLimitOrderTranche(&tick)
+	tranche := NewLimitOrderTrancheWrapper(trancheRaw)
```

Comment:
Not for this PR but since there is a direct correspondence between data tranches and the domain object tranches we shouldn't really need a wrapper for this.

Reply:
Would love to move all of this directly to the LimitOrderTranche type. The issue is that we can't define new types on the LimitOrderTranche type in the keeper package. This makes it very hard to write a Save() method. A couple options I've considered and don't love:
1.) Define save inside the type package and add a  interface shim for Keeper so it can be passed in.
2.) Create SaveLiquidity fn in keeper package. This requires unpacking the underlying type from the interface so it can be passed into the correct KV setter method. 

@teddyknox tagging you on this one since its still of interest